### PR TITLE
Makes iscritter macro check for /obj/critter

### PR DESCRIPTION
**Pull request by pali6**
## About the PR 
`ismobcritter` checks for mob critters, `iscritter` should check for regular critters. This PR makes that happen. It also fixes two minor bugs - welder special attack not hitting objcritters properly and projectiles hitting objcritters playing the obj hit sound instead of mob hit sound.

### CODE REVIEW

File: code/modules/holiday/halloween.dm [hunk 1]
```diff
@@ -45,8 +45,8 @@
 /obj/joeq/spooky
 	name = "Memorial Plaque"
 
-	examine(mob/user)
-		boutput(usr, "Here lies [user.real_name]. Loved by all. R.I.P.")
+	get_desc(dist, mob/user)
+		. += "Here lies [user.real_name]. Loved by all. R.I.P."
```
Comment (Gerhazo) [inline code/modules/holiday/halloween.dm:49]: Possibly should be a separate commit outside the PR for cleaner organisation?

Comment (pali6) [reply code/modules/holiday/halloween.dm]: Yep, my bad.
